openedx/features/courseware/views.py

```diff
@@ -0,0 +1,79 @@
+"""
+Views to add features in courseware.
+"""
+
+from django.utils.translation import ugettext as _
+
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.decorators import (api_view,
+                                       authentication_classes,
+                                       permission_classes)
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework import status
+
+from courseware.courses import get_course_with_access
+from opaque_keys.edx.keys import CourseKey
+
+from lms.djangoapps.courseware.exceptions import CourseAccessRedirect
+from lms.djangoapps.courseware.courseware_access_exception import \
+                                                CoursewareAccessException
+from lms.djangoapps.grades.course_grade_factory import CourseGradeFactory
+from .constants import PRE_ASSESSMENT_FORMAT, POST_ASSESSMENT_FORMAT
+
+
+@api_view()
+@authentication_classes([SessionAuthentication])
+@permission_classes([IsAuthenticated])
```

Comment:
```@view_auth_classes(is_authenticated=True)```
I found this in /edx-platform/lms/djangoapps/courseware/module_render.py
I'm not sure if this is directly applicable here can you check?